### PR TITLE
Java 21

### PR DESCRIPTION
**Pull request by tristantarrant**
This PR ensures that CI builds use JDK 21 (currently EA 18). This allows us to build virtual threads support into `infinispan-commons` and doesn't require the `--enable-preview` flag anymore.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -6,7 +6,7 @@ pipeline {
     }
 
     parameters {
-        choice(name: 'TEST_JDK', choices: ['Default', 'JDK 11', 'JDK 17', 'JDK 20'], description: 'The JDK used to run tests')
+        choice(name: 'TEST_JDK', choices: ['Default', 'JDK 11', 'JDK 17', 'JDK 20', 'JDK 21'], description: 'The JDK used to run tests')
```
Comment (ryanemerson) [inline Jenkinsfile:9]: Should remove "JDK 11" as an option?

File: server/tests/src/test/java/org/infinispan/server/extensions/ExtensionsIT.java [hunk 2]
```diff
@@ -31,14 +32,7 @@ public class ExtensionsIT {
...
    public static String[] mavenArtifacts() {
-      return new String[]{
-            "org.openjdk.nashorn:nashorn-core:15.4",
-            "org.ow2.asm:asm:9.4",
-            "org.ow2.asm:asm-analysis:9.4",
-            "org.ow2.asm:asm-commons:9.4",
-            "org.ow2.asm:asm-tree:9.4",
-            "org.ow2.asm:asm-util:9.4"
-      };
+      return Common.NASHORN_DEPS;
```
Comment (ryanemerson) [inline server/tests/src/test/java/org/infinispan/server/extensions/ExtensionsIT.java:35]: Why is this change necessary?

Comment (tristantarrant) [reply server/tests/src/test/java/org/infinispan/server/extensions/ExtensionsIT.java:35]: Because we already had NASHORN_DEPS in Common, and I'd rather centralize stuff

File: Jenkinsfile (hunk 1 above)
Comment (tristantarrant) [reply Jenkinsfile:9]: Sure
